Packs/Neosec/Integrations/Neosec/Neosec.py

```diff
@@ -0,0 +1,416 @@
+        client: NeosecClient,
+        node_client: Optional[NeosecNodeClient],
+        max_results: int,
+        last_run: Dict[str, int],
+        first_fetch_time: int,
+        alert_status: Optional[str] = None,
+        severities: Optional[List[str]] = None,
+        alert_type: Optional[List[str]] = None,
+) -> Tuple[Dict[str, int], List[dict]]:
+    # Get the last fetch time, if exists
+    # last_run is a dict with a single key, called last_fetch
+    last_fetch = last_run.get("last_fetch", None)
+    # Handle first fetch time
+    if last_fetch is None:
+        # if missing, use what provided via first_fetch_time
+        last_fetch = first_fetch_time
+    else:
+        # otherwise use the stored last fetch
+        last_fetch = int(last_fetch)
+
+    # for type checking, making sure that latest_created_time is int
+    latest_created_time = cast(int, last_fetch)
+
+    # Initialize an empty list of incidents to return
+    # Each incident is a dict with a string as a key
+    incidents: List[Dict[str, Any]] = []
+
+    alerts = client.search_alerts(
+        alert_types=alert_type,
+        alert_status=alert_status,
+        max_results=max_results,
+        start_time=last_fetch,
+        severities=severities,
+    )
+    alerts = detokenize_alerts(node_client, alerts) if node_client else alerts
+
+    for alert in alerts:
+        # If no created_time set is as epoch (0). We use time in ms so we must
+        # convert it from the Neosec API response
+        alert_trigger_at = dateparser.parse(alert["triggered_at"])
+        if not alert_trigger_at:
+            raise ValueError("Alert triggered at is not valid")
+        incident_created_time = alert_trigger_at.timestamp()
+
+        # to prevent duplicates, we are only adding incidents with creation_time > last fetched incident
+        if last_fetch:
+            if incident_created_time <= last_fetch:
+                continue
+
+        incident = {
+            "name": alert["name"],
+            "occurred": timestamp_to_datestring(
+                neosec_datetime_to_timestamp(alert["timestamp"]) * 1000,
+                date_format=DATE_FORMAT,
+                is_utc=True,
+            ),
+            "rawJSON": json.dumps(alert),
+            "severity": convert_to_demisto_severity(alert.get("severity", "Low")),
+            "dbotMirrorId": str(alert["id"]),
+            "details": alert["description"],
+        }
+
+        incidents.append(incident)
+
+        # Update last run and add incident if the incident is newer than last fetch
+        if incident_created_time > latest_created_time:
+            latest_created_time = int(incident_created_time)
+
+    # Save the next_run as a dict with the last_fetch key to be stored
+    next_run = {"last_fetch": latest_created_time}
+    return next_run, incidents
+
+
+def set_alert_status(client: NeosecClient, alert_id: str, alert_status: str) -> None:
+    client.patch_alert(alert_id, alert_status)
+
```

Comment:
return a proper response to the user to indicate if the action was successful. something like -
```
    res = client.patch_alert(alert_id, alert_status)
    if res.status == 200:
        return_results("Alert {alert_id} updated successfully")
    else:
        return_error(f"Error updating alert {alert_id} - {res.content}")
```